.github/workflows/heroku-deploy-dev-ts.yml

```diff
@@ -0,0 +1,53 @@
+name: Deploy backend to Heroku dev environment
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "backend/typescript/**"
+
+jobs:
+  deploy:
+    if: ${{ false }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set Heroku login credentials
+        run: |
+          cat > ~/.netrc <<EOF
+            machine api.heroku.com
+              login $HEROKU_EMAIL
+              password $HEROKU_API_KEY
+            machine git.heroku.com
+              login $HEROKU_EMAIL
+              password $HEROKU_API_KEY
+          EOF
+        env:
+          HEROKU_API_KEY: "${{ secrets.HEROKU_API_KEY }}"
+          HEROKU_EMAIL: "${{ secrets.HEROKU_EMAIL }}"
+      - name: Add Heroku git remote
+        run: heroku git:remote --app $HEROKU_APP_NAME
```

Comment:
Heroku CLI is pre-installed on GitHub Actions runners
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md